plugins/module_utils/vm.py

```diff
@@ -307,8 +310,11 @@ def create_clone_vm_payload(
         *,
         preserve_mac_address,
         source_nics,
+        snapshot_label,
     ):
-        data = {"template": {}}
+        data = dict(template=dict())
+        if snapshot_label:
+            data["snapUUID"] = snapshot_label
```

Comment:
I think you should split this PR into two smaller PRs - one is about vm_snapshot module, how to create and delete them, second is about how to use VM snapshots - clone new VMs from them etc.

The marked line,s I also needed same thing in #218. I just assigned also you to review.

Reply:
Good idea, will do 👍